lookup/main/lookup.go

```diff
@@ -93,6 +93,46 @@ func AddBackend(service string, ip []byte, healthCheckType int) {
 	g.services[newService] = info
 }
 
+var healthCheckMap = map[string]int {
+	"none": healthCheckNone,
+	"http": healthCheckHTTP,
+}
+
+func AddBackendsFromConfig(file string) common.Config {
+	cfg := common.ReadConfig(file)
+	for _, bCfg := range cfg.Backends {
+		healthCheckType, ok := healthCheckMap[bCfg.HealthCheck]
+		if !ok {
+			panic("Unrecognized health check type in config " + bCfg.HealthCheck)
+		}
+		AddBackend(bCfg.Address, bCfg.Ip, healthCheckType)
+	}
+	return cfg
+}
+
+// The common.Config return value can't be exported so unfortunately we need a
+// separate function.
+
+//export AddBackendsFromConfigVoid
+func AddBackendsFromConfigVoid(file string) {
+	AddBackendsFromConfig(file)
+}
+
+// Since Go is garbage-collected and we want to export this function and have
+// Spike (Lua code, effectively C FFI for our concerns) call it to get config
+// args, we have to explicitly convert our return values to C strings
+// explicitly that aren't GC'd (Go does a runtime check and panics if we try to
+// return Go pointers or similar, including Go strings). It's not pretty.
+
+// Also I don't think Go FFI can export Go structs yet, so we're just returning
+// five unnamed values at once...
```

Comment:
Yeah, this whole thing is kinda ugly.  I'm not sure what the best solution is right now, but we can always change it later.